### ISSUE TITLE
Spurious Error Installing into Environment w/ View

### ISSUE THREAD

**Issue by citibeth** [spack/spack#12323]
I have a Spack environment (with view).  I did the command:
```
spack install py-tables
```
It came back to me with:
```
==> Successfully installed py-tables
  Fetch: 0.70s.  Build: 2m 12.17s.  Total: 2m 12.88s.
[+] /root/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.4.0/py-tables-3.4.4-gxdk7c5kh3bgbwcjeqwidzifqsfvch2l

==> Updating view at /root/spack/var/spack/environments/research/.spack-env/view
==> Error: Cannot deactivate r-rcppeigen@0.3.3.3.1%gcc@7.4.0 arch=linux-ubuntu18.04-x86_64/76ff53y because r-car@2.1-4%gcc@7.4.0 arch=linux-ubuntu18.04-x86_64/7kucd5b is activated and depends on it
```
Yes, I have a bunch of R stuff in the environment.  But why is it trying to deactivate / reactive R things when all I did was add `py-tables`?

**Comment by adamjstewart** [comment]
I had this same problem. When I tried to deactivate the package, it told me it wasn't activated to begin with.

**Comment by scheibelp** [comment]
I can look at this by tomorrow (8/9) or possibly later today.

@adamjstewart did you have this issue with `py-tables` specifically, or with some other Python extension (and your error message mentioned an R package)?

@citibeth could you add the output after rerunning with `-d --stacktrace` (like `spack -d --stacktrace install py-tables`)?

**Comment by adamjstewart** [comment]
@scheibelp My error had nothing to do with R packages, it was Python packages that I had already installed. Basically, I installed a set of packages in my environment, one of them crashed, I reconcretized and tried to install again, and it complained that `py-numpy` couldn't be deactivated because `py-scikit-learn` depended on it (both of these were packages I was trying to install). So they weren't random packages, but they were packages that I would like Spack to automatically deactivate and reinstall.

**Comment by scheibelp** [comment]
Can you also include your `spack.yaml`?

**Comment by adamjstewart** [comment]
My `spack.yaml`:
```yaml
spack:
  specs:
  - 'gdal+python'
  - 'python@3:+sqlite3'
  - 'py-earthengine-api'
  - 'py-flake8'
  - 'py-matplotlib'
  - 'py-numpy'
  - 'py-pandas'
  - 'py-pytest@3.9:'
  - 'py-scikit-learn'
  - 'py-scipy@0.14:'
  - 'py-tables'
  - 'py-torch@0.4:'
  - 'py-torchvision'
```

**Comment by citibeth** [comment]
I'm sorry, the workplace where I encountered the problem is not one from which I can share details freely, I've already shared what I can about this problem.  Suffice it to say, Spack has made it beyond academic / government HPC computing!

**Comment by adamjstewart** [comment]
@scheibelp I ran into this problem again today:
```
==> Error: Cannot deactivate py-atomicwrites@1.3.0%clang@11.0.0-apple arch=darwin-catalina-x86_64/u6upsq7 because py-pytest@5.2.1%clang@11.0.0-apple arch=darwin-catalina-x86_64/7wzfzlm is activated and depends on it
```

**Comment by adamjstewart** [comment]
@tgamblin ran into this problem again today on develop. I had an environment already installed (circa when we were working on the performance issues). I decided to update that environment (there is a new version of numpy out).
```console
$ spack concretize -f
...
==> Updating view at /Users/Adam/.spack/darwin/.spack-env/view
==> Error: Cannot deactivate py-scipy@1.4.1%clang@11.0.0-apple arch=darwin-catalina-x86_64/zf2rpjn because py-scikit-learn@0.22%clang@11.0.0-apple+openmp arch=darwin-catalina-x86_64/pbjvmij is activated and depends on it
```
Presumably `py-scipy`, `py-scikit-learn`, and `py-numpy` need to be deactivated since a new version of `py-numpy` (which these depend on) will be installed. Spack is not trying to deactivate packages in the correct order if it's trying to deactivate `py-scipy` before `py-scikit-learn`. I think I know how to reproduce this now if you want to investigate.

**Comment by tgamblin** [comment]
@scheibelp: can you look at this one if you have some free cycles?

**Comment by scheibelp** [comment]
> it complained that py-numpy couldn't be deactivated because py-scikit-learn depended on it (both of these were packages I was trying to install). So they weren't random packages, but they were packages that I would like Spack to automatically deactivate and reinstall.

Thanks for the updates Adam. This appears to be related to (and fixed by): https://github.com/spack/spack/pull/12961

**Comment by scheibelp** [comment]
@adamjstewart I have merged #12961, would you be able to confirm whether it resolves this issue for you?

**Comment by adamjstewart** [comment]
Thanks @scheibelp, I didn't even notice that PR. I'll close this for now and let you know if it happens again.
